archive/utils_archive/filter_activities.py

```diff
@@ -0,0 +1,48 @@
+from utils.Recording import Recording
+import pandas as pd
+
+
+def filter_short_activities(recordings: 'list[Recording]', threshhold: int = 3, strategy: int = 0) -> 'list[Recording]':
+    """
+    Replaces activities shorter than threshhold by [value]. [value] depends on strategy.
+    strategy 0: replaces short activities with previous activity
+    strategy 1: replaces short activities with 'null-activity'
+
+    threshold: number of seconds
+    """
+
+    if strategy != 0 and strategy != 1:
+        raise ValueError('strategy has to be 0 or 1')
+    
+    for recording in recordings:
+        indizes = []
+        length = recording.activities.shape[0]
+
+        for i in range(length-1):
+            if recording.activities.iloc[i] != recording.activities.iloc[i+1]:
+                indizes.append(i+1)
+        
+        for i in range(len(indizes) - 1):
+            if recording.time_frame[indizes[i+1]] - recording.time_frame[indizes[i]] < (threshhold * 1000000):
```

Comment:
Same magic here. Instead of looping and then if-checking, lets first filter the indices and then only loop over the relevant ones:

```python
short_activity_index = np.where(indices[1:] - indices[:-1] < (threshhold * 1000000))[0]
```

(this code is untested, but it should work similar)